### ISSUE TITLE
TypeError

### ISSUE THREAD

**Issue by ValentinBossi** [ticarpi/jwt_tool#21]
When running python3 jwt_tool.py <token> -T or -R i get:

Traceback (most recent call last):
  File "jwt_tool.py", line 1393, in <module>
    tamperToken(paylDict, headDict, sig)
  File "jwt_tool.py", line 761, in tamperToken
    comparestamps, expiredtoken = dissectPayl(paylDict, count=True)
  File "jwt_tool.py", line 1187, in dissectPayl
    elif type(castInput(paylDict[claim][subclaim])) == str:
  File "jwt_tool.py", line 91, in castInput
    newInput = json.loads(newInput)
  File "C:\***\***\AppData\Local\Programs\Python\Python38-32\lib\json\__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not list

or

Traceback (most recent call last):
  File "jwt_tool.py", line 1483, in <module>
    rejigToken(headDict, paylDict, sig)
  File "jwt_tool.py", line 1280, in rejigToken
    comparestamps, expiredtoken = dissectPayl(paylDict)
  File "jwt_tool.py", line 1187, in dissectPayl
    elif type(castInput(paylDict[claim][subclaim])) == str:
  File "jwt_tool.py", line 91, in castInput
    newInput = json.loads(newInput)
  File "C:\***\***\AppData\Local\Programs\Python\Python38-32\lib\json\__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not list

**Comment by ticarpi** [comment]
Could you provide a sample token that causes this behaviour? Thanks.

**Comment by fredsibcald** [comment]
Hey,
I just ran into this same issue when my jwt contained a nested dict object like:

Payload = {
  "context" : {
    "user" : {
      "name" : "Name"
    },
    "group" : "Group"
  },


Switching line 91 in castInput from
newInput = json.loads(newInput)
to
newInput = json.dumps(newInput)

Fixed the issue. (Not sure if it will break further along from this, but it works for my current use-case).

Hope that helps.
